added Project.bpm_at_marker to retrieve the bpm at marker position

IngoHenkel opened the pull request:
-added Project.bpm_at_marker
-updated documentation

IngoHenkel committed:
added Project.bpm_at_marker to retrieve the bpm at marker position

RomeoDespres commented:
Hey @IngoHenkel, thanks for the PR!

I have two small concerns here. First, if you're ok with it, I'd prefer to name this property `bpm_at_cursor` than `bpm_at_marker`, because it is not clear which marker we're talking about here (there can be several in a project, while there can only be one cursor).

Second, `Master_GetTempo()` seems to only work for the currently selected project tab. So in order for it to work consistently even if its project is not currently selected, the call should be wrapped in `Project.make_current_project`. See for instance the definition of [`Project.mute_all_tracks`](https://github.com/RomeoDespres/reapy/blob/730627cee6f39fc26d6ebc8a3df0112e5921cd9f/reapy/core/project/project.py#L708):

```python
@reapy.inside_reaper()
def mute_all_tracks(self, mute=True):
    with self.make_current_project():
        RPR.MuteAllTracks(mute)
```

Note that `@reapy.inside_reaper()` is important here because three API calls are made in this function: one to set project as currently selected, one to mute tracks and one to set previously selected project as currently selected. We should probably do the same here.

IngoHenkel commented:
Hey @RomeoDespres  , ok , that sounds reasonable, i will prepare the changes.
Actually, I wonder why i named it "bpm_at_marker", because all the time I hat "at_cursor" in mind but somehow confused it.

IngoHenkel committed:
implemented review feedback

IngoHenkel commented:
OK, so I implemented it and it seems to work with multiple tabs. Do you want me to squash the commits on my own, or will you do it when merging?

IngoHenkel committed:
fixed wrong action IS for soloing Track

IngoHenkel committed:
fix marker bugs

IngoHenkel committed:
Fix bug with marker indexes

when region markers exist, marker indexes are not straightforward (as "region" markers will get indexes assigned to them along with regular markers).

therefore, get_enum_index in the Marker-Class does not work when regions exist. Fixed that.

IngoHenkel committed:
add methods measure_to_time and time_to_measure to project

IngoHenkel committed:
witch to pyproject.toml

IngoHenkel committed:
port to python 3.14
